common/djangoapps/student/models.py

```diff
@@ -152,10 +152,13 @@ def anonymous_id_for_user(user, course_id, save=True):
     into e.g. personalized survey links.
 
     If user is an `AnonymousUser`, returns `None`
+    else If this user/course_id pair already has an anonymous id in AnonymousUserId object, return that
+    else: create new anonymous_id, save it in AnonymousUserId, and return anonymous id
 
```

Comment:
[idea] Unrelated to diff, but this might be as good a time as any to mention that `course_id` is allowed to be `None` when the call is not in the context of a course.

Reply:
I changed the docs to reflect this possiblity. Please re-review the wording.